### PR TITLE
Implement Residue Partitioning

### PR DESCRIPTION
**Pull request by morrowcj**
Adds the functionality of the residue partition module to the `Field` class.

### CODE REVIEW

File: SC_redesign/Crop_and_Soil/soil/soil_data.py [hunk 1]
```diff
@@ -184,8 +184,10 @@ class SoilData:
     """plant lignin to nitrogen ratio (unitless)"""
     plant_residue_metabolic_fraction: float = 0
     """plant residue fraction that is metabolic (unitless)"""
+    new_residue = 0
+    """amount of new plant material added to the field during the most recent harvest event (kg/ha)"""
     total_residue = 0
-    """"amount of total residue (kg/ha)"""
+    """"amount of total plant material added to the soil from all all cuttings (kg/ha)"""
```
Comment (Isgotkowitz) [inline SC_redesign/Crop_and_Soil/soil/soil_data.py]:
```suggestion
    """"amount of total plant material added to the soil from all cuttings (kg/ha)"""
```

**Review by Isgotkowitz — APPROVED** [review]
Caught 1 typo, everything else looks good.

File: SC_redesign/Crop_and_Soil/field/field.py [hunk 2]
```diff
@@ -138,6 +139,13 @@ def amend_soil(self) -> None:
         self.soil.fertilizer_phosphorus.add_fertilizer_phosphorus(0)
         return
 
+    def add_residue_to_soil(self) -> None:
+        """add residue from the cut crops to the soil"""
+        self.soil.data.new_residue = 0
+        for crop in self.crops:
+            self.soil.data.new_residue += crop.data.yield_residue
+            self.soil.residue_partitioning.partition_residue(crop)
```
Comment (matthew7838) [inline SC_redesign/Crop_and_Soil/field/field.py]: `partition_residue()` needs rainfall and crop as its argument

Comment (morrowcj) [reply SC_redesign/Crop_and_Soil/field/field.py]: Good catch, thank you!

**Review by matthew7838 — APPROVED** [review]

**Comment by morrowcj** [comment]
I'm going to leave this PR unmerged until `CarbonCycling` is completed, since it may impact this feature.

**Comment by morrowcj** [comment]
This PR ended up being unnecessary

**Pull request closed without merging**